include/zephyr/net/wifi_mgmt.h

```diff
@@ -103,14 +115,34 @@ struct wifi_connect_req_params {
 	uint8_t *psk;
 	uint8_t psk_length; /* Min 8 - Max 64 */
 
+	uint8_t *sae_password; /* Optional with fallback to psk */
+	uint8_t sae_password_length; /* No length restrictions */
+
+	uint8_t band;
 	uint8_t channel;
 	enum wifi_security_type security;
+	enum wifi_mfp_options mfp;
+	int timeout; /* -1 - no timeout */
```

Comment:
As discussed offline, it would be good to reflect the timeout in the error code passed along with connect results in connection callback.

Reply:
For now, the result is just `int status`, so, should work for timeout as well, in future PR we can add an enum for that.